actin/personalization/fetch_tool_from_registry

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e
+
+function do_curl() {
+    # New versions (>7.52.1) of `curl` work properly with `--oauth2-bearer` instead of the explicit setting of a header
+    # At time of writing the imager VM was using this older version without prospect of upgrade
+    token="$(gcloud auth print-access-token)"
+    curl -H "authorization: Bearer ${token}" "$@"
+}
+
+AR_URL="https://europe-west4-maven.pkg.dev/actin-build/build-registry-maven/com/hartwig/actin"
+
+[[ $# -ne 3 ]] && echo "USAGE: $0 [tool name] [tool version] [destination path]" && exit 1
+
+tool=$1
+version=$2
+destination=$3
+
+remoteDir="${AR_URL}/${tool}/system/${version}"
+
+expected_md5="$(do_curl -o - -L "${remoteDir}/system-${version}.jar.md5")"
+do_curl -o "${destination}" -L "${remoteDir}/system-${version}.jar"
+actual_md5="$(md5sum "${destination}" | awk '{print $1}')"
+[[ $actual_md5 == "" ]] && echo "No local md5" && exit 1
+if [[ $actual_md5 != "$expected_md5" ]]; then
+    echo "Checksum not as expected or version not found in Artifact Registry for ${tool} version ${version}"
```

Comment:
these are two quite different issues to be combined in one error message. Also wouldn't a not found result in a 404 error to curl?

Reply:
See above tome.

Reply:
@nedleitch are you saying there's no easy way to differentiate between these scenarios? I'm ok with leaving this error as-is, but I guess we could check if `$expected_md5` looks like a valid md5 if we're willing to accept some additional complexity.

Reply:
I'm quite certain that the curl will fail on a 404 before you get to any of the md5 checking. So I would suggest remove the "or version not found" portion of this message to prevent red herrings.

Reply:
I don't think you'll get a 404 if there is a problem with the auth. It wasn't really important to know exactly what the issue was so I admit this was a bit of a shortcut but it fit the original requirements to make sure all failure scenarios were handled.

I agree there will be a 404 if the version is missing.

Checking for a valid md5 won't really help because you will always get a valid one back from `md5sum`. But if something has gone wrong the "JAR" will actually be an error string, so you'll be getting the hash of that string instead of the JAR file. It's a robust way of handling uncertainty but the messaging isn't as specific as desired in this case.

Reply:
I meant that I could check the result of:
```
expected_md5="$(do_curl -o - -L "${remoteDir}/system-${version}.jar.md5")"
```
not the output of `md5sum`.

A "Checksum not as expected" error message in the case where we download an error string instead of a jar file seems accurate enough to me.

Reply:
Of course, sorry didn't read very carefully there.

Easy to check against a regex indeed, I believe `[a-f0-9]{32}` is about the measure of it, should only be two lines.